renderer.php

```diff
@@ -328,4 +328,27 @@ public function presignedurl_tests_content($fs, $testfiles) {
 
         return $output;
     }
+
+    /**
+     * Returns a presigned_url for a file, if the file's extension is whitelisted it returns a local url.
+     * @param $fs
+     * @param \stored_file $file
+     * @param array $headers
+     * @return string
+     * @throws dml_exception
+     */
+    public function generate_presigned_url($fs, $file, array $headers = []) {
+        $filename = $file->get_filename();
+        if ($fs->is_extension_whitelisted($filename)) {
+            return \moodle_url::make_pluginfile_url(
```

Comment:
This feels wrong to be here. I suspect the only reason you need this is because of the testing page, but normal files shouldn't need this. For normal files you should not have gotten this far because the white list would have kicked in earlier.

The problem is that different plugins call moodle_url::make_pluginfile_url in different ways, so this code path while it may work for the test page, isn't 100% correct if called from other paths of moodle. I'd prefer it to be safe and simply return false or null or throw an exception, and it's up to the calling page to default back to the original path.

Reply:
this is used only for the test page, why would it be called from somewhere else?

Reply:
You've subtly changed the public contract. If the url cannot be signed, then in other code it throws an exception and you should probably do the same here. Do we do not every want code to be able to call this and get a url back when it shouldn't. If it did you are just burying a bug that is going to bite someone in the future.

Reply:
the contract is still the same, this is only used for the test page it isn't used.
This is the public method that is used anywhere else: https://github.com/catalyst/moodle-tool_objectfs/blob/master/classes/local/store/object_file_system.php#L755 so the current behaviour remains the same.